libs/vertexai/langchain_google_vertexai/chat_models.py

```diff
@@ -398,25 +397,24 @@ def _generate(
                 project=self.project,
                 convert_system_message_to_human=self.convert_system_message_to_human,
             )
-            message = history_gemini.pop()
-            self.client = _get_client_with_sys_instruction(
+            self.client = _get_client(
                 client=self.client,
                 system_instruction=system_instruction,
                 model_name=self.model_name,
                 safety_settings=safety_settings,
             )
-            with telemetry.tool_context_manager(self._user_agent):
-                chat = self.client.start_chat(history=history_gemini)
 
             # set param to `functions` until core tool/function calling implemented
             raw_tools = params.pop("functions") if "functions" in params else None
             tools = _format_tools_to_vertex_tool(raw_tools) if raw_tools else None
+            tool_config = params.pop("tool_config") if "tool_config" in params else None
```

Comment:
btw, @baskaryan @efriis do you see any issues with pasing a proto directly over the Runnable / chain?
or should it always be a dict, and casted to proto here?

Reply:
If dicts are supported, that would be preferred on the input/output here. Can have a backup type of a proto, but I think serialization will get weird in e.g. LangServe stream log and stream events.